server/py/framework/db/sqldb/db.py

```diff
@@ -5995,6 +5994,77 @@ def list_alerts(
             alerts.append(alert)
         return alerts
 
+    def list_and_delete_project_alerts(
+        self, session, project: str, chunk_size: int = 100
+    ) -> list[str]:
+        """
+        List all alert IDs associated with the specified project and delete them,
+        along with their related notifications, while ensuring foreign key constraints are respected.
+
+        Steps:
+        1. Retrieve all alerts (IDs and names) for the given project.
+        2. Delete related notifications first (since they have foreign key constraints).
+        3. Delete the alerts from the database using ORM-based deletion to ensure cascading works.
+        4. Commit everything at once to improve performance and maintain transactional integrity.
+
+        :param session: SQLAlchemy session for database connection.
+        :param project: Project identifier for which alerts need to be listed and deleted.
+        :param chunk_size: Number of records to delete in each batch (default is 100).
+
+        :return: List of deleted alert IDs.
+        """
+
+        alert_ids = []
+        offset = 0
+
+        while True:
+            # Step 1: Retrieve alerts (IDs and names) for the given project in chunks
+            alerts = (
+                session.query(AlertConfig.id, AlertConfig.name)
+                .filter(AlertConfig.project == project)
+                .limit(chunk_size)
+                .offset(offset)
+                .all()
+            )
+
+            if not alerts:
+                break  # Exit the loop if there are no more alerts to delete
+
+            # Step 2: Delete related notifications before deleting alerts
+            # Notifications reference alerts via foreign keys (parent_id and name).
+            # Deleting them first prevents integrity errors.
+            for alert_id, alert_name in alerts:
+                self._delete_alert_notifications(
+                    session=session,
+                    name=alert_name,  # Index on name improves deletion efficiency
+                    alert_id=alert_id,
+                    project=project,
+                    commit=False,  # Defer commit for batch efficiency
+                )
+
+            # Step 3: Extract alert IDs for deletion and delete alerts in this chunk
+            alert_ids_chunk = [alert_id for alert_id, _ in alerts]
+            alert_ids.extend(alert_ids_chunk)  # Collect all alert IDs to be deleted
+
+            # Step 4: Perform ORM-based deletion for alerts in the current chunk
+            alerts_to_delete = (
+                session.query(AlertConfig)
+                .filter(AlertConfig.id.in_(alert_ids_chunk))
+                .all()
+            )
+            for alert in alerts_to_delete:
+                # Deleting via ORM ensures cascading works
+                session.delete(alert)
```

Comment:
The for loop creates N delete statements to be executed once you commit. You may do it with a single statement instead.
```suggestion
     session.query(AlertConfig)
                .filter(AlertConfig.id.in_(alert_ids_chunk))
                .delete()
```

Reply:
@alonmr actually it doesn't work as expected:

```>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) FOREIGN KEY constraint failed
E       [SQL: DELETE FROM alert_configs WHERE alert_configs.id IN (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
E       [parameters: (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)]
```

Reply:
@alonmr @TomerShor I'm returning the old version of the code. Will glad to hear better suggestions :)

Reply:
What foreign key constraint fails? the notifications, alert states?

Reply:
@TomerShor notifications

Reply:
Why do we need to delete the notifications manually?
The alert config table has a `cascade` on the notification relationship, so once a config is deleted - all its notifications should be deleted too.

Reply:
@TomerShor good question. It was in the old code, I just moved it around. Not sure what was the idea behind. Will remove it

Reply:
Will deleting with `AlertConfig.id in alert_ids` still fail now when you don't delete the notifications?

Reply:
@TomerShor yes

Reply:
The problem is you don't have DB level cascade - see https://github.com/mlrun/mlrun/pull/6598
If you create a query to delete the notifications you can then delete the alerts with 2 queries instead of N but there is a race condition here so IMO leave it as is and add a TODO to fix the foreign key constraint